File: java/ProcedimientosEjercicio2/src/procedimientosejercicio2/ProcedimientosEjercicio2.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Main.java to edit this template
 */
package procedimientosejercicio2;

import java.util.Scanner;

/**
 *
 * @author sdariza
 */
public class ProcedimientosEjercicio2 {

    Scanner input = new Scanner(System.in);

    /**
     * Lectura de una matriz
     *
     * @param A: variable matriz
     * @param n: # de filas y columnas
     */
    void readMatrix(int A[][], int n) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                System.out.printf("\nIngrese la pos [%d,%d]\n", i, j);
                A[i][j] = input.nextInt();
            }
        }
        System.out.println("");
    }

    /**
     * Imprime la matrix
     *
     * @param A: Variable matriz
     * @param n: dimension cuadrada
     */
    void printMatrix(int A[][], int n) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                System.out.print(A[i][j] + "|");
            }
            System.out.println("");
        }
        System.out.println("");
    }

    /**
     * Método para sumar dos matrices: A+B = C
     *
     * @param A: Variable matriz
     * @param B: Variable Matriz
     * @param n: Dimensión de las matrices
     * @param C: Variable matriz
     * @param signo: Determina la suma o la resta
     */
    void sumarRestarMatriz(int A[][], int B[][], int n, int C[][], int signo) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                C[i][j] = A[i][j] + signo * B[i][j];
            }
        }
    }

    /**
     * Método para multiplicar matrices
     *
     * @param A: Variable matriz
     * @param B: Variable matriz
     * @param n: dimensión de la matriz
     * @param C : Matriz con el producto AB=C
     */
    void multiplicarMatrices(int A[][], int B[][], int n, int C[][]) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                for (int k = 0; k < n; k++) {
                    C[i][j] = C[i][j] + A[i][k] * B[k][j];
                }
            }
        }
    }

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        // TODO code application logic here
        ProcedimientosEjercicio2 ob = new ProcedimientosEjercicio2();
        int A[][], B[][], C[][], D[][], n, E[][], F[][];
        System.out.println("Ingrese la dimensión de las matrices");
        n = ob.input.nextInt();
        A = new int[n][n];
        B = new int[n][n];
        C = new int[n][n];
        D = new int[n][n];
        E = new int[n][n];
        F = new int[n][n];
        System.out.println("Leer matriz A");
        ob.readMatrix(A, n);
        System.out.println("Leer matriz B");
        ob.readMatrix(B, n);
        System.out.println("Matriz A");
        ob.printMatrix(A, n);
        System.out.println("Matriz B");
        ob.printMatrix(B, n);
        ob.sumarRestarMatriz(A, B, n, C, 1);
        System.out.println("A+B=C");
        ob.printMatrix(C, n);
        ob.multiplicarMatrices(C, A, n, D);
        ob.sumarRestarMatriz(A, B, n, E, -1);
        ob.multiplicarMatrices(E, B, n, F);
        System.out.println("(A+B)A");
        ob.printMatrix(D, n);
        System.out.println("(A-B)B");
        ob.printMatrix(F, n);
    }

}

File: python/ProcedimientosEjercicio2.py
import numpy as np


def read_matrix(A: np.array, n:int):
    """ Módulo para leer una matriz con numpy

    Args:
        A (np.array): Variable matriz
        n (int): dimensión de la matriz - cuadrada
    """
    for i in range(0,n,1):
        for j in range(0,n,1):
            A[i][j] = int(input(f'Ingrese el valor de la pos {i,j}'))
        

def print_matrix(A: np.array, n:int):
    """ Módulo para imprimir una matriz con numpy

    Args:
        A (np.array): Variable matriz
        n (int): dimensión de la matriz - cuadrada
    """
    for i in range(0,n,1):
        for j in range(0,n,1):
            print(A[i][j], end='|')
        print("")
        

def sum_res_matrices(A:np.array, B:np.array, n:int, C:np.array, signo:int):
    """Módulo para sumar o restar dos matrices A - B = C

    Args:
        A (np.array): variable matriz
        B (np.array): variable matriz
        n (int): dimensión de las matrices
        C (np.array): A-B = C: variable matriz
        signo (int): para sumar: 1 , restar: -1
    """
    for i in range(n):
        for j in range(n):
            C[i][j] = A[i][j] + signo * B[i][j]


def mul_matrix(A:np.array, B:np.array, n:int, C:np.array):
    """_summary_

    Args:
        A (np.array): _description_
        B (np.array): _description_
        n (int): _description_
        C (np.array): _description_
    """
    for i in range(n):
        for j in range(n):
            for k in range(n):
                C[i][j] = C[i][j] + A[i][k] * B[k][j]


if __name__ == "__main__":
    n = int(input('Ingrese el valor de la dimensión de las matrices: '))
    # print(np.array[[1,2,3],[1,34,5]]) # {{1,2,4}}
    A = np.zeros((n,n), dtype=int)
    B = np.zeros((n,n), dtype=int)
    C = np.zeros((n,n), dtype=int)
    D = np.zeros((n,n), dtype=int)
    E = np.zeros((n,n), dtype=int)
    F = np.zeros((n,n), dtype=int)
    print("Lectura de la matriz A")
    read_matrix(A,n)
    print("Lectura de la matriz B")
    read_matrix(B,n)
    print("Matriz A:")
    print_matrix(A,n)
    print("Matriz B:")
    print_matrix(B,n)
    sum_res_matrices(A,B,n,C,1)
    mul_matrix(C,A,n,D)
    sum_res_matrices(A,B,n,E,-1)
    mul_matrix(E,B,n,F)
    print("(A+B)A")
    print_matrix(D,n)
    print("(A-B)B")
    print_matrix(F,n)
